lib/routes/universities/cug/gcxy/index.js

```diff
@@ -70,40 +63,40 @@ module.exports = async (ctx) => {
                 },
             });
             const $ = cheerio.load(response.data);
-            const items = $('div.right_list > ul > li');
+            const rowList = $('ul.col-news-list > li.list_item');
+            // get all items from the rowList
+            const items = rowList
+                .map((index, item) => {
+                    const li = $(item);
+                    return {
+                        title: li.find('a.news-title').text(),
+                        link: new URL(li.find('a.news-title').attr('href'), host).href,
+                        description: '',
+                        pubDate: new Date(li.find('span.news-date').text()).toUTCString(),
+                    };
+                })
+                .get();
+
             return Promise.all(
-                items
-                    .map(async (index, item) => {
-                        item = $(item);
-                        const a = item.find('a');
-                        const link = host + a.attr('href').replace('..', '');
-                        // console.log(link);
-                        return {
-                            title: a.attr('title'),
-                            link,
-                            description: await got({
-                                method: 'get',
-                                url: link,
-                                headers: {
-                                    Referer: host,
-                                },
-                            })
-                                .then((res) => {
-                                    const $ = cheerio.load(res.data);
-                                    return $('.v_news_content').html();
-                                })
-                                .catch(() => ''),
-                            pubDate: item.find('span').text(),
-                        };
+                items.map((element) =>
+                    ctx.cache.tryGet(`cug/${element.link}`, async () => {
+                        try {
+                            const result = await got.get(String(element.link));
```

Comment:
https://github.com/DIYgod/RSSHub/blob/4fea5c788a5bf1c63f97122e930cd463421c87ee/lib/routes/universities/cug/gcxy/index.js#L73
`.href` returns a `String`. Don't recast the same type.
docs: https://nodejs.org/api/url.html#urlhref
https://docs.rsshub.app/joinus/script-standard.html

Reply:
Please explain how recasting `element.link` **from `string` to `string`** can help converting relative path to absolute path.